### PR TITLE
docs: Migrate the overview page to Bigframes official landing page

### PR DESCRIPTION
**Pull request by ashleyxuu**
The staging page for the new overview page on the API doc: https://screenshot.googleplex.com/8MPN5LMoLu2CGCB

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -13,411 +13,10 @@ BigQuery DataFrames is an open-source package. You can run
...
-------------------------
-
-BigQuery DataFrames is designed for scale, which it achieves by keeping data
-and processing on the BigQuery service. However, you can bring data into the
-memory of your client machine by calling ``.to_pandas()`` on a DataFrame or Series
-object. If you choose to do this, the memory limitation of your client machine
-applies.
-
-
-License
```
Comment (tswast) [inline README.rst:404]: Please keep this information. It's necessary to know that we have third-party code when folks download our package.

Comment (ashleyxuu) [reply README.rst:404]: Done.

File: README.rst [hunk 2]
```diff
@@ -10,395 +10,14 @@ powered by the BigQuery engine.
 BigQuery DataFrames is an open-source package. You can run
 ``pip install --upgrade bigframes`` to install the latest version.
 
+
 Documentation
 -------------
 
+* `Use BigQuery DataFrames <https://cloud.google.com/bigquery/docs/use-bigquery-dataframes>`_
```
Comment (tswast) [inline README.rst:17]: This is a broken link.

Comment (tswast) [reply README.rst:17]: I recommend the quickstart: https://cloud.google.com/bigquery/docs/dataframes-quickstart

File: README.rst [hunk 3]
```diff
@@ -10,395 +10,19 @@ powered by the BigQuery engine.
 BigQuery DataFrames is an open-source package. You can run
 ``pip install --upgrade bigframes`` to install the latest version.
 
+
```
Comment (tswast) [inline README.rst:13]: Let's add a getting started link:

```suggestion
Getting started with BigQuery DataFrames
----------------------------------------

Try the [BigQuery DataFrames quickstart](https://cloud.google.com/bigquery/docs/dataframes-quickstart) to get up and running in just a few minutes.

```

Comment (tswast) [reply README.rst:13]: Nevermind, I see it below.

**Review by tswast — APPROVED** [review]

**Pull request merged**